app/portainer/services/api/endpointService.js

```diff
@@ -54,10 +54,15 @@ function EndpointServiceFactory($q, Endpoints, FileUploadService) {
     return Endpoints.remove({id: endpointID}).$promise;
   };
 
-  service.createLocalEndpoint = function() {
+  service.createLocalEndpoint = function(Windows) {
```

Comment:
Please rename the parameter to `useNamedPipe`.

Reply:
Hmm. If we assume ```npipe://``` or ```unix:///``` based on OS then we can remove this parameter and whole Windows/Linux containers selection.

I will change it like that.

Reply:
Yes, if Portainer is running on Windows, then creating a local endpoint should default to use npipe. If running on Linux, default to use unix socket.

This is totally doable in the backend as we can create platform specific files (see my comment about `local_windows.go` and `local_linux.go` above).

But the frontend is not aware of the platform where Portainer is running. that's why #1186 introduced a `Local` boolean parameter in the endpoint creation payload.

Reply:
OK. I will re-check the original implementation